### PR TITLE
feat: another pass at Node.js bindings

### PR DESCRIPTION
**Pull request by dignifiedquire**
TODOs

- [x] finish porting tests
- [x] fixup CI: macos universal builds
- [x] fixup CI: publish step
- [x] cleanup

### CODE REVIEW

File: iroh-js/src/doc.rs [hunk 1]
```diff
@@ -0,0 +1,1302 @@
...
+        Ok(res)
+    }
+
+    /// Get sync peers for this document
+    #[napi]
+    pub async fn get_sync_peers(&self) -> Result<Option<Vec<Vec<u8>>>> {
+        let list = self.inner.get_sync_peers().await?;
+        let list = list.map(|l| l.into_iter().map(|p| p.to_vec()).collect());
+        Ok(list)
+    }
```
Comment (b5) [inline iroh-js/src/doc.rs:381]: is the value that comes back from this the bytes of the public key? Do I need to base32 these myself up in JS land?

File: README.md [hunk 2]
```diff
@@ -11,6 +11,9 @@ This repo defines Python, Swift, and Kotlin bindings for [iroh](https://github.c
 [Swift: SwifPM](https://swiftpackageindex.com/n0-computer/iroh-ffi)
 [Rust: Crates](https://crates.io/crates/iroh)
 
+Comming Soon
```
Comment (Arqu) [inline README.md:14]: We should probably fill this in too.

Comment (dignifiedquire) [reply README.md]: fixed

**Review by Arqu — APPROVED** [review]
LGTM 😭

**Pull request merged**